### PR TITLE
[wip] Switch project to yii2-bootstrap4

### PR DESCRIPTION
**Pull request by machour**
Ported this template to yii2-bootstrap4, in order to test the new extension.

So far so good except that the `template` option doesn't seem to be taken in account anymore.

### CODE REVIEW

File: views/site/login.php [hunk 1]
```diff
@@ -29,7 +29,8 @@
         <?= $form->field($model, 'password')->passwordInput() ?>
 
         <?= $form->field($model, 'rememberMe')->checkbox([
-            'template' => "<div class=\"col-lg-offset-1 col-lg-3\">{input} {label}</div>\n<div class=\"col-lg-8\">{error}</div>",
+            // FIXME: NOPE not showing
+            'template' => "<div class=\"col-lg-offset-1 col-lg-3\">NOPE {input} {label}</div>\n<div class=\"col-lg-8\">{error}</div>",
```
Comment (machour) [inline views/site/login.php]: @simialbi any idea why `template` isn't working? Did the syntax change or is something broken?

**Comment by samdark** [comment]
@machour do you think we should ship application with BS4 from next release?

**Comment by machour** [comment]
@samdark I do think it would make sense to have it as a default for new apps. We could conduct a quick survey on the forum to get more opinions.

**Comment by samdark** [comment]
https://forum.yiiframework.com/t/should-yii-2-applications-switch-to-bootstrap-4-for-new-projects/124717

File: views/site/login.php (hunk 1 above)
Comment (simialbi) [reply views/site/login.php]: The offset syntax changed in bootstrap 4: `col-lg-offset-1` in bs3 is `offset-lg-1` in bs4:;
see https://getbootstrap.com/docs/4.2/layout/grid/#offsetting-columns

Comment (simialbi) [reply views/site/login.php]: https://github.com/yiisoft/yii2-bootstrap4/pull/100

**Comment by samdark** [comment]
Fix merged. @machour would you please retest?

**Comment by machour** [comment]
@simialbi tried your fix, there's still a little glitch:

Code:

```php
<?php $form = ActiveForm::begin([
        'id' => 'login-form',
        'layout' => 'horizontal',
        'fieldConfig' => [
            'template' => "{label}\n<div class=\"col-lg-3\">{input}</div>\n<div class=\"col-lg-8\">{error}</div>",
            'labelOptions' => ['class' => 'col-lg-2 control-label'],
        ],
]); ?>

    [...]
    <?= $form->field($model, 'rememberMe')->checkbox([
        'template' => "<div class=\"offset-lg-1 col-lg-3\">{input} {label}</div>\n<div class=\"col-lg-8\">{error}</div>",
    ]) ?>

   [...]
<?php ActiveForm::end(); ?>
```

| Bootstrap 3 | Bootstrap 4 |
|------------|--------------|
| screen shot 2019-01-21 at 9 44 09 am|screen shot 2019-01-21 at 9 44 42 am|

It seems that with bootstrap3, using the `template` option on a field discarded the Form's `labelOptions`, which seems normal. 

This is not the case anymore with bootstrap4, and we end-up with a broken bootstrap hierarchy (a col inside a col)

**Comment by simialbi** [comment]
@machour this is beacause of the custom template. With default `horizontal` layout without customizing the template it renders all correct

**Comment by machour** [comment]
No changes with the array syntax, `col-lg-2 control-label` still gets applied to the label when it should not.
screen shot 2019-01-21 at 10 43 53 am

**Comment by simialbi** [comment]
I'll check that... Ty

**Comment by machour** [comment]
We came the conclusion that merging this PR would do more harm than good.

**Pull request closed without merging**